app/src/main/java/tech/relaycorp/gateway/domain/publicsync/DeliverPublicParcels.kt

```diff
@@ -0,0 +1,85 @@
+package tech.relaycorp.gateway.domain.publicsync
+
+import kotlinx.coroutines.flow.asFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.filter
+import kotlinx.coroutines.flow.flatMapLatest
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.take
+import tech.relaycorp.gateway.common.Logging.logger
+import tech.relaycorp.gateway.data.database.StoredParcelDao
+import tech.relaycorp.gateway.data.disk.DiskMessageOperations
+import tech.relaycorp.gateway.data.disk.MessageDataNotFoundException
+import tech.relaycorp.gateway.data.model.RecipientLocation
+import tech.relaycorp.gateway.data.model.StoredParcel
+import tech.relaycorp.gateway.domain.DeleteParcel
+import tech.relaycorp.gateway.domain.LocalConfig
+import tech.relaycorp.gateway.pdc.PoWebClientBuilder
+import tech.relaycorp.poweb.PoWebClient
+import tech.relaycorp.relaynet.bindings.pdc.ClientBindingException
+import tech.relaycorp.relaynet.bindings.pdc.RejectedParcelException
+import tech.relaycorp.relaynet.bindings.pdc.ServerException
+import tech.relaycorp.relaynet.bindings.pdc.Signer
+import tech.relaycorp.relaynet.cogrpc.readBytesAndClose
+import java.util.logging.Level
+import javax.inject.Inject
+
+@JvmSuppressWildcards
+class DeliverPublicParcels
```

Comment:
I'd suggest not to use the term "public parcels" because that sounds like the endpoint is public; e.g., `https://api.twitter.com/relaynet`. How about something like `DeliverParcelsToGateway`, for example?

Same applies to `CollectPublicParcels`.